src/pages/confirmation/[id].tsx

```diff
@@ -75,19 +76,48 @@ export const getServerSideProps: GetServerSideProps = async (
   context: GetServerSidePropsContext
 ) => {
   const deviceType = deviceDetect(context.req.headers["user-agent"] ?? "");
+  
+  // Extract id from URL params
+  const { id:bookingID  } = context.params as { id: string };
+  // Extract request_code from query parameters
+  const { response_code } = context.query;
```

Comment:
Same here is context.query is undefined then destructuring will probably cause a runtime error.

Better use optional chaining:

const response_code = context.query?.response_code;